File: axon-form-core/internal/edge/edge.go
package edge

type Edge struct {
	ID         string            `json:"id"`
	Label      string            `json:"label"`
	SourceNode string            `json:"source_node"`
	TargetNode string            `json:"target_node"`
	Type       EdgeType          `json:"type"`
	Conditions []EdgeCondition   `json:"conditions"`
	Configs    map[string]string `json:"config"`
}

type EdgeCondition struct {
	ID            string              `json:"id"`
	Edge          string              `json:"edge"`
	CheckNode     string              `json:"node"`
	Expression    ConditionExpression `json:"expr"`
	ExpectedValue string              `json:"value"`
}

type EdgeConditionGroup struct {
	ID          string `json:"id"`
	Conditions  string `json:"conditions"`
	NodeID      string `json:"node"`
	EdgeIDs     []string
	PostfixExpr []string
	ValidEdges  map[string]bool
}

File: axon-form-console/src/types/Edge.ts
import type { EdgeType } from '@/configs/graph';
import { parseNodeCondition, type NodeCondition } from './Node';
import type { EdgeResponse } from './PocketBaseResponse';

export interface Edge {
  id: string;
  label: string;
  sourceNode: string;
  targetNode: string;
  edgeType: EdgeType;
  conditions: NodeCondition[];
}

export function parseEdgeResponse(edge: EdgeResponse): Edge {
  return {
    id: edge.id,
    label: edge.label,
    sourceNode: edge.source_node,
    targetNode: edge.target_node,
    edgeType: edge.type as EdgeType,
    conditions: edge.expand?.conditions_via_edge?.map(parseNodeCondition) ?? [],
  };
}
